app/controllers/claims/support/schools_controller.rb

```diff
@@ -6,4 +6,44 @@ def index
   def show
     @school = Claims::School.find(params.require(:id))
   end
+
+  def new
+    @school = Claims::School.new
+  end
+
+  def check
+    if school.valid? && !school.claims?
+      @school = school.decorate
+    else
+      school.errors.add(:urn, :taken) if school.claims?
+      render :new
+    end
+  end
+
+  def create
+    school.claims = true
+    if school.save
+      redirect_to claims_support_schools_path
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def school
+    @school ||=
+      begin
+        gias_school = GiasSchool.find_by(urn: urn_param)
+        if gias_school.blank?
+          Claims::School.new
+        else
+          gias_school.school || gias_school.build_school
+        end
+      end
+  end
+
+  def urn_param
+    params.dig(:gias_school, :urn) || params.dig(:school, :urn)
+  end
```

Comment:
This also raises the question, on the `SchoolsController`, should our request body ever be `gias_school`?

This standardises body params pattern, which _should_ help with consistency:

```suggestion
  def school_params
    params.require(:school).permit(:urn)
  end
  
  def urn_param
    school_params.require(:urn)
  end
```

Not strongly opinionated about the `school_params` part, but I don't think we need to support `params[:gias_school]`.

Reply:
The reason for the original implementation is to account for the two different forms.
- check (Finding the GiasSchool to be used to create/check against a school)
- create (Creating a new School record using the `:urn`)

Reply:
Got it. Makes sense :) I'm wondering now if `check` belongs in a `GiasSchoolsController`, but no action needed. Just a thought 🤔 .

All good! Thanks!